src/mon/PGMap.cc

```diff
@@ -4550,8 +4556,41 @@ int reweight::by_utilization(
       osd_util.second = (double)p.second.kb_used / (double)p.second.kb;
     }
     util_by_osd.push_back(osd_util);
+
+    // Overweighted devices and their weights are factors to
+    // calculate reweight_urgency. One 10% underfilled device
+    // with 5 2% overfilled devices, is arguably a better
+    // situation than one 10% overfilled with 5 2% underfilled devices
+
+    if ( osd_util.second >= average_util){
+      unsigned weight = osdmap.get_weight(p.first);
+      sum_weights += weight;
+
+      /*
+       Have a function F(x), where x = (osd_util.second -
+       average_util) / average_util, which is bounded between 0 and 1, so
+       that ultimately reweight_urgency will also be bounded. A larger value
+       of x, should imply more urgency to reweight. Also, the difference
+       between F(x) when x is large, should be minimal. The value of F(x)
+       should get close to 1 (highest urgency to reweight) with steeply.
+       Could have used F(x) = (1 - e^(-x)). But that had slower
+       convergence to 1, compared to the function currently in use.
+      */
+
+      // cdf of normal distribution:https://stackoverflow.com/a/18786808
```

Comment:
@amitkumar50 Would a [Wikipedia link](https://en.wikipedia.org/wiki/Error_function#Cumulative_distribution_function) for the same be okay?